build: enable typescript for cypress

ktmud opened the pull request:
### SUMMARY

Follow up #10161 :

- Enable TypeScript for Cypress
- Convert support files to TS
- Convert an example test (dashboard/filter.test.js) to TS
- Reenable Prefer TS check for `cypress-base`

<strike>Currently `cypress-base` is now using a newer version of TypeScript than `superset-frontend`. Will file another PR to upgrade later.</strike>

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

All Cypress tests should still be valid and pass.


cc @etr2460

ktmud committed:
build: enable typescript for cypress

Follow up #10161 :

- Enable TypeScript for Cypress
- Convert support files to TS
- Convert an example test (dashboard/filter.test.js) to TS
- Upgrade TypeScript for both `cypress-base` and `superset-frontend`
- Reenable Prefer TS check for `cypress-base`

ktmud commented on superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts:
```diff
@@ -72,7 +85,7 @@ describe('Dashboard filter', () => {
 
     cy.get('.Select__control input[type=text]')
       .first()
-      .focus({ force: true })
```
Type check helped by notifying me this is not a valid API for `focus`.

ktmud commented on superset-frontend/cypress-base/cypress/integration/dashboard/filter.test.ts:
```diff
@@ -32,13 +44,14 @@ describe('Dashboard filter', () => {
+      filterId =
+        dashboard.slices.find(
+          slice => slice.form_data.viz_type === 'filter_box',
+        )?.slice_id || 0;
```
Fallback to 0 is easier than running assertions on a potential `undefined`.

etr2460 commented:
Looks like CI is still breaking a bit, will review once fixed (i think you're missing installing an import)

ktmud committed:
Disable a eslint error

Some modules might not be available when `eslint` in the parent folder
before running `npm install` in the cypress folder.

ktmud commented on superset-frontend/cypress-base/.eslintrc:
```diff
@@ -0,0 +1,25 @@
+    "plugin:cypress/recommended"
+  ],
+  "rules": {
+    "import/no-unresolved": 0,
```
Some Cypress modules might not be available if you run `eslint` in the parent folder before `npm install` in `cypress-base` (which is what our CI does).

ktmud commented:
@etr2460 Fixed the tests and this is now ready for review.

etr2460 reviewed (approved):
lgtm, thanks for taking the time to do this!
